### ISSUE TITLE
Issue with ReorderableListView.builder: 'package:flutter/src/widgets/overlay.dart': Failed assertion: line 720 pos 12: 'constraints.biggest.isFinite': is not true.

### ISSUE THREAD

**Issue by jrmallorca** [flutter/flutter#79940]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Surround a ReorderableListView.builder with a Column or
   copy this file:

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  List<int> numbers = <int>[1, 2, 3];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          ReorderableListView.builder(
            shrinkWrap: true,
            itemCount: numbers.length,
            itemBuilder: (context, index) {
              final number = numbers[index];
              return ListTile(
                  key: ValueKey(number), title: Text(number.toString()));
            },
            onReorder: (oldIndex, newIndex) {
              final number = numbers.removeAt(oldIndex);
              numbers.insert(newIndex, number);
            },
          ),
        ],
      ),
    );
  }
}
```
</details>

**Expected results:** A list view.

**Actual results:** No list view.

<details>
  <summary>Logs</summary>

flutter run -v
```
           ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
           The following assertion was thrown during performResize():
           'package:flutter/src/widgets/overlay.dart': Failed assertion: line 720 pos 12:
           'constraints.biggest.isFinite': is not true.

           Either the assertion indicates an error in the framework itself, or we should provide substantially
           more information in this error message to help you determine and fix the underlying cause.
           In either case, please report this assertion by filing a bug on GitHub:
             https://github.com/flutter/flutter/issues/new?template=2_bug.md

           The relevant error-causing widget was:
             ReorderableListView file:///home/jrmallorca/Projects/bug/lib/main.dart:42:31

           When the exception was thrown, this was the stack:
           #2      _RenderTheatre.computeDryLayout (package:flutter/src/widgets/overlay.dart:720:12)
           #3      RenderBox.performResize (package:flutter/src/rendering/box.dart:2342:12)
           #4      RenderObject.layout (package:flutter/src/rendering/object.dart:1763:9)
           #5      ChildLayoutHelper.layoutChild (package:flutter/src/rendering/layout_helper.dart:54:11)
           #6      RenderFlex._computeSizes (package:flutter/src/rendering/flex.dart:830:43)
           #7      RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:932:32)
           #8      RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #9      MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:171:12)
           #10     _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:925:7)
           #11     MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:243:7)
           #12     RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:407:14)
           #13     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #14     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #15     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #16     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #17     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1371:11)
           #18     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #19     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #20     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #21     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #22     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #23     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #24     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #25     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #26     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #27     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #28     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #29     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #30     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #31     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #32     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #33     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #34     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:3362:13)
           #35     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #36     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #37     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #38     _RenderTheatre.performLayout (package:flutter/src/widgets/overlay.dart:743:15)
           #39     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #40     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #41     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #42     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #43     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #44     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #45     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #46     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #47     RenderCustomPaint.performLayout (package:flutter/src/rendering/custom_paint.dart:546:11)
           #48     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #49     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #50     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #51     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:118:14)
           #52     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
           #53     RenderView.performLayout (package:flutter/src/rendering/view.dart:153:14)
           #54     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1641:7)
           #55     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:884:18)
           #56     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:453:19)
           #57     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:874:13)
           #58     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:319:5)
           #59     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1144:15)
           #60     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1082:9)
           #61     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:865:7)
           (elided 13 frames from class _AssertionError, class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)

           The following RenderObject was being processed when the exception was fired: _RenderTheatre#405b9 NEEDS-LAYOUT NEEDS-PAINT
           NEEDS-COMPOSITING-BITS-UPDATE:
             creator: _Theatre ← Overlay ← ReorderableListView ← Column ← _BodyBuilder ← MediaQuery ←
               LayoutId-[<_ScaffoldSlot.body>] ← CustomMultiChildLayout ← AnimatedBuilder ← DefaultTextStyle ←
               AnimatedDefaultTextStyle ← _InkFeatures-[GlobalKey#8a98c ink renderer] ← ⋯
             parentData: offset=Offset(0.0, 0.0); flex=null; fit=null (can use size)
             constraints: BoxConstraints(0.0<=w<=360.0, 0.0<=h<=Infinity)
             size: MISSING
             skipCount: 0
             textDirection: ltr
           This RenderObject had the following descendants (showing up to depth 5):
               onstage 1: RenderPadding#b8d16 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
                 child: RenderRepaintBoundary#89447 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
                   child: RenderCustomPaint#acdeb NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
                     child: RenderRepaintBoundary#397d7 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
                       child: _RenderScrollSemantics#e0377 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
               no offstage children
           ════════════════════════════════════════════════════════════════════════════════════════════════════
[ +230 ms]
           Another exception was thrown: BoxConstraints forces an infinite height.
[ +138 ms]
           Another exception was thrown: RenderBox was not laid out: _RenderTheatre#405b9 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
[ +294 ms]
           Another exception was thrown: RenderBox was not laid out: RenderFlex#c0124 relayoutBoundary=up1 NEEDS-PAINT
           NEEDS-COMPOSITING-BITS-UPDATE
```

```
```

```
```

</details>

**Comment by pedromassangocode** [comment]
I cannot reproduce this on latest master.

<img width="324" alt="Screen Shot 2021-04-07 at 13 24 56" src="https://user-images.githubusercontent.com/67424416/113865969-a72b5e00-97a4-11eb-8935-648e4c6fa86e.png">

Can you please provide your `flutter doctor -v`?
Thank you

**Comment by jrmallorca** [comment]
Hello, here is my `flutter doctor -v`

```
[✓] Flutter (Channel stable, 2.0.4, on Linux, locale en_GB.UTF-8)
    • Flutter version 2.0.4 at /opt/flutter
    • Framework revision b1395592de (6 days ago), 2021-04-01
      14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK
    version 30.0.3)
    • Android SDK at /opt/android-sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /opt/android-sdk
    • ANDROID_SDK_ROOT = /opt/android-sdk
    • Java binary at: /usr/lib/jvm/java-8-openjdk/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_282-b08)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at
    google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome
      executable.

[!] Android Studio (not installed)
    • Android Studio not found; download from
      https://developer.android.com/studio/index.html
      (or visit
      https://flutter.dev/docs/get-started/install/linux#android-se
      tup for detailed instructions).

[✓] Connected device (1 available)
    • SM G900F (mobile) • 755447bf • android-arm • Android 6.0.1
      (API 23)

! Doctor found issues in 2 categories.
```

**Comment by pedromassangocode** [comment]
I was able to reproduce this on stable but is seems to fixed on latest master (see https://github.com/flutter/flutter/issues/79940#issuecomment-814873367).

The solution is to give the list view a constraints, (`SizedBox` or similars).

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 2.1.0-13.0.pre.524, on macOS 11.2.3 20D91 darwin-x64, locale en-AO)
    • Flutter version 2.1.0-13.0.pre.524 at /Users/pedromassango/Code/flutter_master
    • Framework revision 98a9618993 (67 minutes ago), 2021-04-07 22:34:03 -0700
    • Engine revision cf86823971
    • Dart version 2.13.0 (build 2.13.0-213.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/pedromassango/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 54.1.4
    • Dart plugin version 203.7759

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • AE2F4F51-880E-44FB-BB29-171B766D9B54 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.
```
```bash
[✓] Flutter (Channel stable, 2.0.4, on macOS 11.2.3 20D91 darwin-x64, locale en-AO)
    • Flutter version 2.0.4 at /Users/pedromassango/Code/flutter_stable
    • Framework revision b1395592de (6 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/pedromassango/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 54.1.4
    • Dart plugin version 203.7759

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • AE2F4F51-880E-44FB-BB29-171B766D9B54 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.
```

</details>

**Comment by Piinks** [comment]
> Not sure if this should be closed as fixed.
> Cc @Piinks

@darrenaustin is the right person to say so on this I think.

**Comment by darrenaustin** [comment]
Yes, this was fixed with #79024. See [this comment](https://github.com/flutter/flutter/issues/76150#issuecomment-807861195) for a possible work around until it makes it to stable.
